source/gui.py

```diff
@@ -0,0 +1,332 @@
+            self.pallet_elements.append(shapes.Circle(           loc.Box(x=0, y= 6*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            self.pallet_elements.append(shapes.HalfCircle(       loc.Box(x=0, y= 7*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            self.pallet_elements.append(shapes.Square(           loc.Box(x=0, y= 8*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            self.pallet_elements.append(shapes.Heart(            loc.Box(x=0, y= 9*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            self.pallet_elements.append(shapes.Star(             loc.Box(x=0, y=10*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            self.pallet_elements.append(shapes.SymmetricTriangle(loc.Box(x=0, y=11*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            
+            for shape in self.pallet_elements:
+                shape.draw_shape(self, location_offset=loc.Pos())
+            self.update()
+        redraw_pallet_elements()
+        def pick_up(event):
+            if (self.verbose_events): print(f'<pick_up> at {event.x},{event.y}')
+            pallet_item = get_pallet_item(loc.Pos(event.x, event.y))
+
+            if pallet_item != PalletItem.NONE:
+                if PalletItem.YELLOW.value <= pallet_item.value <= PalletItem.BLUE.value: # It is an color
+                    self.last_color = pallet_item
+                    redraw_pallet_elements()
+                self.in_hand.append(pallet_item)
+                if (self.verbose_events): print (f'Picked up {pallet_item.name}')
+                return
+            
+            # Pick_up event did not happen in the pallet
+            event.x -= self.pallet_item_size().x
+            for shape in self.list_of_canvas_shapes:
+                if event.x < shape.box.pos.x: continue                      # Left of shape.box  (out of range)
+                if event.x > shape.box.pos.x + shape.box.size.x: continue   # Right of shape.box (out of range)
+                if event.y < shape.box.pos.y: continue                      # Top of shape.box   (out of range)
+                if event.y > shape.box.pos.y + shape.box.size.y: continue   # Bottom of shape.box(out of range)
+
+                self.in_hand.append(shape)
+                shape.remove_shape(self)
+                self.list_of_canvas_shapes.remove(shape)
+                if (self.verbose_events): print (f'Shape picked up from: {shape.box}')
+                return
+            if (self.verbose_events): print ('No shape in the region')
```

Comment:
no return at the end of the function, just in case?

Reply:
Nope it would not add value.  
See other comment